### ISSUE TITLE
cloud build example to use option for defaultLogsBucketBehavior

### ISSUE THREAD

**Issue by sandygvs** [google-github-actions/setup-gcloud#697]
### TL;DR

Trying to run the example workflow and getting the following error:

terraform version: 1.8.4
GCP provider version : 6.3.0
The build is running, and logs are being written to the default logs bucket. However I'm trying to configure the cloud build to store the logs in user-owned and and regionalized bucket as per https://cloud.google.com/build/docs/securing-builds/store-manage-build-logs#store_build_logs_in_a_user-owned_and_regionalized_bucket

### Expected behavior

Below code should work and it should create regional specific bucket in the respective GCP project and region
```
resource "google_cloudbuild_trigger" "trigger" {
  name = "my-trigger"
  location = "global"

  trigger_template {
    branch_name = "main"
    repo_name   = "my-repo"
  }

  build {
    step {
      name = "ubuntu"
      args = ["-c", "exit 1"]
      allow_failure = true
    }
 .
.
.
.
    }
    options {
      **default_log_bucket_behavior = "REGIONAL_USER_OWNED_BUCKET"**
      worker_pool = "pool"
      }
    }
  }
}
```

### Observed behavior

Error: Unsupported argument
│ 
│   on inf_trigger/main.tf line 93, in resource "google_cloudbuild_trigger" "inf_build_trigger":
│   93:       default_log_bucket_behavior = "REGIONAL_USER_OWNED_BUCKET"
│ 
│ An argument named "default_log_bucket_behavior" is not expected here.

### Action YAML

```yaml
steps:
  - name: >-
      us-docker.pkg.dev/test-common/test-images/builder-image@sha256:08de19772fe6e7776a757c0afa9d0e2717bc88b4142b03f206c79259b32d9e51
    args:
      - '-c'
      - >-
        terraform init --backend-config=../../state/dev.conf && terraform
        validate
    dir: infrastructure/deployments/api
    id: init-terraform-test
    entrypoint: /bin/sh
  - name: >-
      us-docker.pkg.dev/test-common/test-images/builder-image@sha256:08de19772fe6e7776a757c0afa9d0e2717bc88b4142b03f206c79259b32d9e51
    args:
      - '-c'
      - >-
        terraform apply -auto-approve -input=false -parallelism=10
        -var-file=env/dev.tfvars
    dir: infrastructure/deployments/api_compliance
    id: execute-terraform-test
    entrypoint: /bin/sh
timeout: 2000s
options:
  workerPool: projects/test-test/locations/us-central1/workerPools/inf
  defaultLogsBucketBehavior: REGIONAL_USER_OWNED_BUCKET
tags:
  - inf
  - deploy
  - dev
```

### Additional information

The config as mentioned here https://cloud.google.com/build/docs/securing-builds/store-manage-build-logs#store_build_logs_in_a_user-owned_and_regionalized_bucket is working if I update the cloud build.yaml directly but not able to do this via terraform.
